File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        int startIndex = line.find(":") + 2;
        return line.substr(startIndex);
    }

    std::string log_level(std::string line) {
        int startIndex = line.find("[") + 1;
        int endIndex = line.find("]") - 1;
        return line.substr(startIndex, endIndex);
    }

    std::string reformat(std::string line) {
        return message(line) + " (" + log_level(line) + ")";
    }  // namespace log_line
}

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int startIndex = logLine.IndexOf(":")+1;
        return logLine.Substring(startIndex).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int startIndex = logLine.IndexOf("[")+1;
        int endIndex = logLine.IndexOf("]");
        return logLine.Substring(startIndex, endIndex-startIndex).ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}
